server/src/main/java/org/elasticsearch/bootstrap/ElasticsearchUncaughtExceptionHandler.java

```diff
@@ -54,21 +53,11 @@ static boolean isFatalUncaught(Throwable e) {
     void onFatalUncaught(final String threadName, final Throwable t) {
         final String message = "fatal error in thread [" + threadName + "], exiting";
         logger.error(message, t);
-        Terminal.DEFAULT.errorPrintln(message);
-        t.printStackTrace(Terminal.DEFAULT.getErrorWriter());
-        // Without a final flush, the stacktrace may not be shown before ES exits
-        Terminal.DEFAULT.flush();
-
-        Elasticsearch.printLogsSuggestion();
```

Comment:
I have a question about removing the printLogsSuggestion(), do we not want to keep the friendly error message for the user to check the logs for more details? We show the last bit of logs on the screen, but there might be more insight of the error in the logs.

Reply:
printLogsSuggestion() writes to the Terminal. So it's effectively like the others, it might be broken because we've closed stderr.

However, your comment made me realize we are implicitly relying on the uncaught exception handler for startup as well, by allowing main have a checked Exception. I will change that to have a broad try/catch so that we can capture and print the exception ourselves instead of relying on the JVM to do so for us.

Reply:
I pushed 2d14865